src/hooks/use-contract-activity.ts

```diff
@@ -1,42 +1,78 @@
-import { useCallback, useEffect, useState } from 'react';
+import { useEffect, useMemo, useState } from 'react';
 
-import { ActivityGroup, ActivityInterface } from '../interfaces/activity.interface';
-import { OperationLiquidityBakingInterface } from '../interfaces/operation.interface';
+import { ActivityGroup } from '../interfaces/activity.interface';
+import { TokenTypeEnum } from '../interfaces/token-type.enum';
 import { UseActivityInterface } from '../interfaces/use-activity.interface';
 import { useSelectedRpcUrlSelector } from '../store/settings/settings-selectors';
 import { useSelectedAccountSelector } from '../store/wallet/wallet-selectors';
-import { transformActivityInterfaceToActivityGroups } from '../utils/activity.utils';
+import { LIQUIDITY_BAKING_DEX_ADDRESS } from '../token/data/token-slugs';
+import { TEZ_TOKEN_SLUG } from '../token/data/tokens-metadata';
 import { isDefined } from '../utils/is-defined';
-import { mapOperationLiquidityBakingToActivity } from '../utils/operation.utils';
-import { getContractOperations } from '../utils/token-operations.util';
+import { loadLastActivity, LoadLastActivityTokenType } from '../utils/token-operations.util';
+import { useTokenType } from './use-token-type';
 
-export const useContractActivity = (contractAddress: string): UseActivityInterface => {
+export const useContractActivity = (tokenSlug?: string): UseActivityInterface => {
+  const [contractAddress, tokenId] = (tokenSlug ?? '').split('_');
+  const { tokenType, loading } = useTokenType(contractAddress);
   const { publicKeyHash } = useSelectedAccountSelector();
-  const selectedRpcUrl = useSelectedRpcUrlSelector();
+  const selectedRpc = useSelectedRpcUrlSelector();
 
   const [isAllLoaded, setIsAllLoaded] = useState<boolean>(false);
   const [activities, setActivities] = useState<Array<ActivityGroup>>([]);
 
-  const loadLastActivity = useCallback(
-    async (lastLevel: number | null) => {
-      const loadedActivities = await loadContractActivity(selectedRpcUrl, publicKeyHash, contractAddress, lastLevel);
-
-      setIsAllLoaded(loadedActivities.length === 0);
-      const activityGroups = transformActivityInterfaceToActivityGroups(loadedActivities);
-      setActivities(prevValue => [...prevValue, ...activityGroups]);
-    },
-    [publicKeyHash, setActivities]
-  );
+  const activityType = useMemo(() => {
+    if (isDefined(tokenSlug)) {
+      switch (tokenSlug) {
+        case TEZ_TOKEN_SLUG:
+          return LoadLastActivityTokenType.Tezos;
+        case LIQUIDITY_BAKING_DEX_ADDRESS:
+          return LoadLastActivityTokenType.LiquidityBaking;
+        default:
+          return tokenType;
+      }
+    } else {
+      return LoadLastActivityTokenType.All;
+    }
+  }, [tokenType]);
```

Comment:
delete `LoadLastActivityTokenType`, you don't need it.  
make this switch case where you are loading operations hashes